log/format.go

```diff
@@ -177,14 +151,18 @@ func FormatLogfmtValue(value interface{}, term bool) string {
 			return "nil"
 		}
 		return formatLogfmtUint256(v)
+	case error:
+		return v.Error()
+
+	case fmt.Stringer:
+		return v.String()
```

Comment:
So, I suspect that `v` being nil here might `panic`, and we might have to add back that `recover` clause, _or_ check for nil (the tests should tell us). 

However, it's usually not nil, so it might be preferrable to have an expensive recover rather than regularly check for nilness. At least it seems that the implementors of slog have chosen that route.

Reply:
Yup

```
$ go run -tags=integrationtests ./cmd/geth --log.format terminal logtest  
INFO [11-14|20:16:15.800] big.Int                                  111,222,333,444,555,678,999=111,222,333,444,555,678,999
...
INFO [11-14|20:16:15.802] (fmt.Stringer)(nil)                      res=nil
panic: value method time.Time.String called using nil *Time pointer

goroutine 1 [running]:
time.(*Time).String(0x0?)
```

Reply:
Ah, you also moved the `stringer` earlier, so now the `common.Hash` triggers the `String()` instead of `TerminalString()`. I'll push a fix